### PR TITLE
Setup dev environment using docker-compose

### PR DESCRIPTION
**Pull request by arunvelsriram**
- Start the dependencies like backend server and postgres database using docker-compose
- Update readme with dev machine setup instructions

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM node:10
+WORKDIR /opt/app
+COPY package*.json ./
+RUN npm install
```
Comment (vraravam) [inline Dockerfile]: Should this not be `npm install --production`?

Comment (arunvelsriram) [reply Dockerfile]: `ng` is required to build the project which is a dev dependency. We could use multi stage build. I envision this to be replaced by a web server like nginx. What do you think?

Comment (vraravam) [reply Dockerfile]: I agree with moving to nginx.
@subbu3012 - wdyt?

**Comment by vraravam** [comment]
@arunvelsriram - one more thought (but please feel free to rebut it) - can we do separate branches in the main codebase, and raise PRs - rather than in your individual forks? Reason is that it is easier for those of us who review (by just switching branches and diffing locally). It is also easier for us to contribute onto that branch - in case of anything that we see and modify.

File: Dockerfile (hunk 1 above)
Comment (subbu3012) [reply Dockerfile]: @vraravam @arunvelsriram i think i need some context here. what do you want to replace with ngnix?

Comment (vraravam) [reply Dockerfile]: @subbu3012  - the thought process is whether to have `ng` as a runtime dependency for production-mode or whether we can create static assets that can be served via an nginx process.

Comment (subbu3012) [reply Dockerfile]: @vraravam In dev mode, angular uses `webpack-dev-server` internally which is wrapped under `ng serve` command. i dont think we need(and have) this in production. 

In production,`aws-amplify` is taking care of serving the static assets from `dist` directory. 

if you think we have more to discuss on this, we(you, @arunvelsriram, and i) can set up a small call and do so.

**Comment by vraravam** [comment]
@arunvelsriram - can you please also incorporate the migrations command into the Dockerfile?

**Comment by arunvelsriram** [comment]
> @arunvelsriram - one more thought (but please feel free to rebut it) - can we do separate branches in the main codebase, and raise PRs - rather than in your individual forks? Reason is that it is easier for those of us who review (by just switching branches and diffing locally). It is also easier for us to contribute onto that branch - in case of anything that we see and modify.

Yes, it makes sense. Will raise upcoming PRs from a branch in the original repo.

**Comment by arunvelsriram** [comment]
> @arunvelsriram - can you please also incorporate the migrations command into the Dockerfile?

This is done: https://github.com/chennaitricolor/wastexchange-be/pull/10

File: Dockerfile (hunk 1 above)
Comment (arunvelsriram) [reply Dockerfile]: In the call we decided that we will nginx for
1. serving frontend (static website)
2. acting as a proxy for backend

**Comment by arunvelsriram** [comment]
@vraravam is this good to merge? 

I will raise a separate PR for using nginx to serve the frontend app.

**Comment by vraravam** [comment]
LGTM

**Pull request merged**